tests/utils/test_downstream.py

```diff
@@ -0,0 +1,134 @@
+    downstream_datasets = tamr_toolbox.utils.downstream.datasets(dataset)
+    # check that these datasets are included in the list of downstream datasets
+    PIPELINE_DEPENDENT_DATASETS = {
+        "minimal_mastering_unified_dataset_dedup_user_defined_signals",
+        "minimal_golden_records_source_list_diff",
+        "minimal_mastering_unified_dataset_dedup_clusters_union",
+        "minimal_mastering_unified_dataset_dedup_dnf_binning",
+        "minimal_mastering_unified_dataset_dedup_cluster_stats_union",
+        "minimal_golden_records_golden_records_cluster_profile",
+        "minimal_mastering_unified_dataset_dedup_published_cluster_stats",
+        "minimal_mastering_unified_dataset_dedup_pair_comments",
+        "minimal_mastering_unified_dataset_dedup_function_predictions",
+        "minimal_mastering_unified_dataset_important_pairs",
+        "minimal_mastering_unified_dataset_dedup_features",
+        "minimal_golden_records_golden_records",
+        "minimal_mastering_unified_dataset_dedup_clusters_with_stats_union",
+        "minimal_mastering_unified_dataset_dedup_cluster_accuracy_metrics",
+        "minimal_mastering_unified_dataset_dedup_idf",
+        "minimal_mastering_unified_dataset_dedup_labels",
+        "minimal_mastering_unified_dataset_dedup_clusters",
+        "minimal_mastering_unified_dataset_dedup_published_clusters_with_data",
+        "minimal_mastering_unified_dataset_dedup_test_records",
+        "minimal_mastering_unified_dataset_dedup_training_clusters",
+        "minimal_mastering_unified_dataset_dedup_published_clusters",
+        "minimal_mastering_unified_dataset_dedup_test_clusters",
+        "minimal_mastering_unified_dataset_dedup_feedback",
+        "minimal_golden_records_golden_records_draft",
+        "minimal_mastering_unified_dataset_dedup_high_impact_questions",
+        "minimal_mastering_unified_dataset_dedup_test_records_accuracy",
+        "minimal_mastering_unified_dataset_dedup_model",
+        "minimal_golden_records_source_list",
+        "minimal_mastering_unified_dataset_dedup_all_persistent_ids",
+        "minimal_mastering_unified_dataset_dedup_human_signals",
+        "minimal_mastering_unified_dataset_dedup_cluster_similarities",
+        "minimal_mastering_unified_dataset_dedup_non_null_count",
+        "minimal_mastering_unified_dataset_dedup_dnf_signals",
+        "minimal_mastering_unified_dataset_dedup_published_cluster_counts",
+        "minimal_golden_records_golden_records_pinned_cluster_input",
+        "minimal_mastering_unified_dataset_dedup_cluster_stats",
+        "minimal_mastering_unified_dataset_dedup_signals_predictions",
+        "minimal_golden_records_golden_records_rule_output",
+        "minimal_mastering_unified_dataset_dedup_clusters_with_data",
+        "minimal_mastering_unified_dataset_dedup_signals",
+    }
+    assert PIPELINE_DEPENDENT_DATASETS.issubset(set([d.name for d in downstream_datasets]))
```

Comment:
Why use issubset and not check for an exact match?

Reply:
in case someone runs profiling on one of the derived datasets. But we do not run online tests usually so I will change to exact match.